client/extensions/woocommerce/app/promotions/promotion-models.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { translate as moduleTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import CurrencyField from './fields/currency-field';
+import PercentField from './fields/percent-field';
+import TextField from './fields/text-field';
+
+export function createModels( translate ) {
+	// -------------------------
+	// Reusable Promotion Fields
+	// -------------------------
+	const couponCode = {
+		component: TextField,
+		labelText: translate( 'Coupon code' ),
+		explanationText: translate(
+			'Only apply this promotion when the customer supplies the coupon code'
+		),
+		placeholderText: translate( 'Enter coupon code' ),
+		isRequired: true,
+	};
+
+	const fixedDiscount = {
+		component: CurrencyField,
+		isRequired: true,
+	};
+
+	// ---------------
+	// Promotion types
+	// ---------------
+	const productSaleModel = {
+		salePrice: {
+			component: CurrencyField,
+			labelText: translate( 'Product Sale Price' ),
+			isRequired: true,
+		},
+	};
+
+	const fixedProductModel = {
+		couponCode,
+		fixedDiscount: {
+			...fixedDiscount,
+			labelText: translate( 'Product Discount', { context: 'noun' } )
+		},
+	};
+
+	const fixedCartModel = {
+		couponCode,
+		fixedDiscount: {
+			...fixedDiscount,
+			labelText: translate( 'Cart Discount', { context: 'noun' } ),
+		},
+	};
+
+	const percentCartModel = {
+		couponCode,
+		percentDiscount: {
+			component: PercentField,
+			labelText: translate( 'Percent Cart Discount', { context: 'noun' } ),
+			isRequired: true,
+		},
+	};
+
+	// Master list of models by promotion type.
+	return {
+		[ 'product_sale' ]: productSaleModel,
+		[ 'fixed_product' ]: fixedProductModel,
+		[ 'fixed_cart' ]: fixedCartModel,
+		[ 'percent' ]: percentCartModel,
+	};
+}
+
+export default createModels( moduleTranslate );
```

Comment:
Out of curiosity, does this need to be a function passing translate, or can you use translate directly from `i18n-calypso` and export the model object directly?

Reply:
I could use translate directly, totally. But I've always wondered about using the global module translate function--if it will be easy to change out languages and such. But there's so much of Calypso that already does it this way, maybe I shouldn't worry about it.

Reply:
I went ahead and removed the function wrapping the objects and am just using `translate` directly from the import. We do this a lot of other places, and it wouldn't be hard to wrap it up in a function again if needed.